File: Medium/Even Swap/even-swap.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.lang.*;
import java.util.*;

class GFG{
	public static void main(String [] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int test = Integer.parseInt(br.readLine());
		while(test-- > 0) {
			int n = Integer.parseInt(br.readLine());
			int [] arr = new int[n];
			String [] str = br.readLine().trim().split(" ");
			for(int i = 0; i < n; i++) {
				arr[i] = Integer.parseInt(str[i]);
			}
			Solution obj = new Solution();
			int [] b = obj.lexicographicallyLargest(arr, n);
			for(int x: b) {
				System.out.print(x+" ");
			}
			System.out.println();
		}
	}
}
// } Driver Code Ends


//User function Template for Java

class Solution{
	int [] lexicographicallyLargest(int [] arr, int n) {
		 //Write your code here
        boolean flag = true;
        
        while(flag){
            flag = false;
            for(int i=0; i<n-1; i++){
                if((arr[i]+arr[i+1])%2 == 0){
                    if(arr[i] < arr[i+1]){
                        int temp = arr[i+1];
                        arr[i+1] = arr[i];
                        arr[i] = temp;
                        flag = true;
                    }
                }
            }
        }
        return arr;


	}
}

File: Medium/Even Swap/even-swap.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++

class Solution
{
    public:
       vector <int> lexicographicallyLargest(vector <int> &a,int n)
        {
            int prev=0;
            for(int i=1;i<n;i++)
            {
                if(a[i]%2==a[i-1]%2)// odd odd or even even
                    continue;
                else // previous group was a group that can be sorted
                {
                     sort(a.begin()+prev,a.begin()+i,greater<int>{});
                     prev=i; // new group starting here
                }
            }
            sort(a.begin()+prev,a.end(),greater<int>{});
            return a;
        }
};

//{ Driver Code Starts.

int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        vector <int> a(n);
        for(auto &j:a)
            cin>>j;
        Solution s;
        vector <int> b=s.lexicographicallyLargest(a,n);
        for(auto i:b)
            cout<<i<<" ";
        cout<<endl;
    }
    return 0;
}
// } Driver Code Ends